scipy/special/_basic.py

```diff
@@ -2209,6 +2215,12 @@ def comb(N, k, exact=False, repetition=False):
     if repetition:
         return comb(N + k - 1, k, exact)
     if exact:
+        if int(N) != N or int(k) != k:
```

Comment:
```suggestion
        if np.any((np.floor(N) != N) | (np.floor(k) != k)):
```

`comb` supports broadcasting so scalar conversions (`int(k)`) and comparisons (`int(k) != k`) would fail with array inputs

Reply:
h`comb` is only vectorized when 

> `comb` is vectorized and supports broadcasting so scalar conversions (`int(k)`) and comparisons (`int(k) != k`) would fail with array inputs

Currently `comb` is only vectorized if `exact=False`. See the Notes [here](https://github.com/scipy/scipy/blob/c4829bddb859ffe5716a88f6abd5e0d2dc1d9045/scipy/special/_basic.py#L2191-L2195). `_comb_int` can only handle scalar arguments, so changing the check here won't make this work with array inputs.  Also if `comb` were vectorized when `exact=True`, I think the preferred behavior would be to return `nan` for non-integer arguments and display a warning, similar to what happens when you do vectorized division of numpy arrays and zero divisions occur.

Reply:
This does not actually change the current behaviour.

Current main branch:
```

In [1]: from scipy.special import comb

In [2]: comb([4,5],[2,3], exact=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-ec42b3305665> in <module>
----> 1 comb([4,5],[2,3], exact=True)

~/development/scipy/scipy/special/_basic.py in comb(N, k, exact, repetition)
   2211         return comb(N + k - 1, k, exact)
   2212     if exact:
-> 2213         return _comb_int(N, k)
   2214     else:
   2215         k, N = asarray(k), asarray(N)

_comb.pyx in scipy.special._comb._comb_int()

TypeError: int() argument must be a string, a bytes-like object or a number, not 'list'
```
This branch.
```
In [1]: from scipy.special import comb

In [2]: comb([4,5],[2,3], exact=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-ec42b3305665> in <module>
----> 1 comb([4,5],[2,3], exact=True)

~/development/scipy/scipy/special/_basic.py in comb(N, k, exact, repetition)
   2217         return comb(N + k - 1, k, exact)
   2218     if exact:
-> 2219         if int(N) != N or int(k) != k:
   2220             warnings.warn(
   2221                 "Non-integer arguments are deprecated when exact=True "

TypeError: int() argument must be a string, a bytes-like object or a number, not 'list'

```
I am not sure an extra error message is needed?

Reply:
Happy to revise the error message. Do you think it is fine to leave the check as `int(N) != N or int(k) != k` until support for arrays is added?

Reply:
> Do you think it is fine to leave the check as `int(N) != N or int(k) != k` until support for arrays is added?

Yeah, better to keep it as-is; clearly shows arrays aren't supported.

Reply:
> Does this sound good?

+1 for option 2. It's going to be a little tricky to modify this to work with arrays, but I definitely prefer not to box ourselves in by going with option 1.

Reply:
Thanks for the feedback @steppi @j-bowhay! Let's change the deprecation message then.